src/providers/bitcoin/BitcoinSwapProvider.js

```diff
@@ -228,8 +228,14 @@ export default class BitcoinSwapProvider extends Provider {
     const output = initiationTx.outputs[voutIndex]
     const value = parseInt(reverseBuffer(output.amount).toString('hex'), 16)
     const { relayfee } = await this.getMethod('jsonrpc')('getinfo')
-    const fee = this.getMethod('calculateFee')(1, 1, feePerByte)
-    const amount = value - Math.max(fee, BigNumber(relayfee).times(1e8).toNumber())
+    const calculatedFee = this.getMethod('calculateFee')(1, 1, feePerByte)
+    const fee = Math.max(calculatedFee, BigNumber(relayfee).times(1e8).toNumber())
```

Comment:
Since we're doing BigNumber calcs, should we not use BigNumber.max, and then also keep doing the math as value.minus(fee) 

Somehow feel uneasy about mixing BigNumbers with built-in data types

Reply:
Good point! fixed